app/src/main/java/protect/card_locker/Utils.java

```diff
@@ -669,21 +670,32 @@ public static void patchColors(AppCompatActivity activity) {
     // rendering mess
     // use after views are inflated
     public static void postPatchColors(AppCompatActivity activity) {
-        TypedValue typedValue = new TypedValue();
-        activity.getTheme().resolveAttribute(android.R.attr.colorBackground, typedValue, true);
-        activity.findViewById(android.R.id.content).setBackgroundColor(typedValue.data);
+        if (Build.VERSION.SDK_INT <= Build.VERSION_CODES.P) {
+            activity.findViewById(android.R.id.content).setBackgroundColor(resolveBackgroundColor(activity));
+        }
```

Comment:
Please use ints as that's what most of Catima users. If you want to change the standard, let's do that in another MR (and let's start documenting these things somewhere).

And you say "I tested this in the emulator with Android 6 and 13.", I'm most curious actually about the versions that no longer run this code. So, 10 and 11 and 12 and 14 too.

So this needs a lot of testing on multiple different themes (that I don't have time to do right now).

Reply:
> Please use ints as that's what most of Catima users.

Does it? See above.

I'll look into testing more. No rush :)

Reply:
Well, nvm, just more testing then :)

Reply:
Turns out this code is responsible for correctly applying the "Pure black background for dark theme" setting when dark theme is used in the MainActivity. Adding the version check breaks the OLED black on new Android versions (noticed on Android 14)

Strangely enough, this code isn't necessary in the LoyaltyCardEditActivity, what a mess...

Reply:
Will do more testing & debugging. Hopefully this week.

Reply:
Maybe it'd be best to just leave this code on all API versions with a new comment like:

```
// Force correct color
// Fixes OLED dark mode in MainActivity
```

So we at least understand better what it does. And then later debug further